transaction/transaction.go

```diff
@@ -361,6 +363,132 @@ func MakeAssetConfigTxn(account string, feePerByte, firstRound, lastRound uint64
 	return tx, nil
 }
 
+// MakeAssetTransferTxn creates a tx for sending some asset from an asset holder to another user
+// the recipient must have marked themselves as willing to accept the asset
+// - account is a checksummed, human-readable address that will send the transaction and assets
+// - recipient is a checksummed, human-readable address what will receive the assets
+// - closeAssetsTo is a checksummed, human-readable address that behaves as a close-to address for the asset transaction; the remaining assets not sent to recipient will be sent to closeAssetsTo. Leave blank for no close-to behavior.
+// - amount is the number of assets to send
+// - feePerByte is a fee per byte
+// - firstRound is the first round this txn is valid (txn semantics unrelated to asset management)
+// - lastRound is the last round this txn is valid
+// - note is an arbitrary byte array
+// - genesis id corresponds to the id of the network
+// - genesis hash corresponds to the base64-encoded hash of the genesis of the network
+// - creator is the address of the asset creator
+// - index is the asset index
+func MakeAssetTransferTxn(account, recipient, closeAssetsTo string, amount, feePerByte, firstRound, lastRound uint64, note []byte,
+	genesisID, genesisHash, creator string, index uint64) (types.Transaction, error) {
+	var tx types.Transaction
+
+	tx.Type = types.AssetTransferTx
+
+	accountAddr, err := types.DecodeAddress(account)
+	if err != nil {
+		return tx, err
+	}
+
+	ghBytes, err := byte32FromBase64(genesisHash)
+	if err != nil {
+		return types.Transaction{}, err
+	}
+
+	tx.Header = types.Header{
+		Sender:      accountAddr,
+		Fee:         types.MicroAlgos(feePerByte),
+		FirstValid:  types.Round(firstRound),
+		LastValid:   types.Round(lastRound),
+		GenesisHash: types.Digest(ghBytes),
+		GenesisID:   genesisID,
+		Note:        note,
+	}
+
+	creatorAddr, err := types.DecodeAddress(creator)
+	if err != nil {
+		return tx, err
+	}
+
+	tx.XferAsset = types.AssetID{
+		Creator: creatorAddr,
+		Index:   index,
+	}
+
+	recipientAddr, err := types.DecodeAddress(recipient)
+	if err != nil {
+		return tx, err
+	}
+	tx.AssetReceiver = recipientAddr
+
+	if closeAssetsTo != "" {
+		closeToAddr, err := types.DecodeAddress(closeAssetsTo)
+		if err != nil {
+			return tx, err
+		}
+		tx.AssetCloseTo = closeToAddr
+	}
+
+	tx.AssetAmount = amount
+
+	// Update fee
+	eSize, err := estimateSize(tx)
+	if err != nil {
+		return types.Transaction{}, err
+	}
+	tx.Fee = types.MicroAlgos(eSize * feePerByte)
+
+	return tx, nil
+}
+
+// MakeAssetAcceptanceTransaction creates a tx for marking an asset as willing to be accepted by an account
```

Comment:
Is the marking the asset or marking the account? 
This statement does not seem to agree with the other comment: 
"the recipient must have marked themselves as willing to accept the asset"